### PR TITLE
refactor: relayer uses the new solana gateway

### PR DESCRIPTION
**Pull request by roberts-pumpurs**
**Describe the changes**

Closes [#588](https://github.com/eigerco/solana-axelar-internal/issues/558)

- Task processor for Solana will use the new ixs and the new encoding when interacting with the Solana Gateway
- Solana event parser will use the new event types from the new Gateway
- misc: updated deps & deny.toml (some deps of our deps are no longer maintained but there's nothing we can do about that)

**Checklist**

- [x] Tests have been added/updated for the changes.
- [x] Documentation has been updated for the changes (if applicable).
- [x] The code follows Rust's style guidelines.

### CODE REVIEW

File: crates/solana-gateway-task-processor/src/component.rs [hunk 1]
```diff
@@ -147,511 +144,225 @@ async fn process_task(
...
+    // For compatibility reasons with the rest of the Axelar protocol we need add custom handling
+    // for ITS & Governance programs
+    let destination_address = message.destination_address.parse::<Pubkey>()?;
+    match destination_address {
+        axelar_solana_its::ID => {
+            // todo ITS specific handling
+            tracing::error!("ITS program not yet supported");
+        }
+        axelar_solana_governance::ID => {
+            // todo Governance specific handling
```
Comment (roberts-pumpurs) [inline crates/solana-gateway-task-processor/src/component.rs:209]: @eloylp this is for this issue -- https://github.com/eigerco/axelar-relayer-core-internal/issues/28

File: crates/solana-gateway-task-processor/src/component.rs [hunk 2]
```diff
@@ -147,511 +144,225 @@ async fn process_task(
...
-        while let Some(result) = future_set.next().await {
-            let pubkey = result?;
-            command_accounts.push(pubkey);
+            // if security passed, we broadcast the tx
+            let ix = axelar_executable::construct_axelar_executable_ix(
+                message,
+                &payload,
+                gateway_incoming_message_pda,
+            )?;
+            send_tx_parse_error(solana_rpc_client, keypair, ix).await?;
```
Comment (roberts-pumpurs) [inline crates/solana-gateway-task-processor/src/component.rs:235]: if the tx cannot be sent (we get an error), this needs to be reported back to the Amplifier API.
https://github.com/eigerco/solana-axelar-internal/issues/504

**Review by frenzox — APPROVED** [review]

File: crates/solana-event-forwarder/src/component.rs [hunk 3]
```diff
@@ -348,182 +239,18 @@ fn map_gateway_event_to_amplifier_event(
                     )
                     .build(),
             );
-            tracing::info!(message_id = ?approved_message.message_id, "Message approved");
+            tracing::info!("message approved");
             Some(amplifier_event)
         }
-        MessageExecuted(ref _executed_message) => {
+        GatewayEvent::MessageExecuted(ref _executed_message) => {
```
Comment (roberts-pumpurs) [inline crates/solana-event-forwarder/src/component.rs:245]: This is the part for https://github.com/eigerco/solana-axelar-internal/issues/572 issue to be implemented

File: crates/solana-gateway-task-processor/src/component.rs (hunk 1 above)
Comment (eloylp) [reply crates/solana-gateway-task-processor/src/component.rs:209]: ack, thank you

**Pull request merged**